app/services/post_alerter.rb

```diff
@@ -356,6 +356,15 @@ def create_notification(user, type, post, opts = {})
       end
     end
 
+    # Don't alert same user about the same post
+    # i.e. skip duplicate alerts on a revised post (when editing post triggers a new type of notification like a mention or a quote)
+    existing_alert = user.notifications
```

Comment:
It seems to me you could query this once, and use it for the other check too, since it's a subset of this one. (You could use ruby code to check for the type in the resulting array.)